### PR TITLE
feature: collateral settings/manager

### PR DESCRIPTION
**Pull request by stackchain**
- [x] add collateral to utxos manager
- [x] set initial state to load collateral
- [x] remove collateral from other utxos when set (affects balance)
- [x] auto assign a pure utxo as collateral
- [x] create a tx to create the collateral
- [x] remove selected collateral
- [ ] check copies with product
- [ ] check suggested layout with product
- [x] check states (with collateral, with collateral consumed, without collateral)
- [x] check state - no balance for collateral (tx creation, balance - deposit >= min collateral)
- [x] check collateral range 5-10 ADA with product (defined 2-5 ADA)
- [x] the yoroi-lib is complaining when sending to own external (missing vkey)
- [x] between the dispatch of unsingedTx and navigation it needs a little delay, the state is not committed when navigating to confirmTx (unsingedTx is empty)
- [x] wrap collateral stories within SendProvider (added later)
- [x] translations

### CODE REVIEW

File: apps/wallet-mobile/src/yoroi-wallets/cardano/utxoManager/utxos.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+) {
+  const findById = (id: RawUtxo['utxo_id']) => {
+    return utxos.find((utxo) => utxo.utxo_id === id)
+  }
+  const exists = (id: RawUtxo['utxo_id']) => {
+    return findById(id) !== undefined
+  }
+
+  const findCollateralCandidates = () => {
+    return utxos.filter(isPureUtxo).filter((utxo) => isAmountInCollateralRange(utxo.amount, {maxLovelace, minLovelace}))
```
Comment (stackchain) [inline apps/wallet-mobile/src/yoroi-wallets/cardano/utxoManager/utxos.ts]: minimal first - sort by utxo.amount (string)

File: apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+const withCollateralRemoveLoading: YoroiWallet = {
+  ...mocks.wallet,
+  setCollateralId: mocks.setCollateralId.loading,
+}
+
+storiesOf('ManageCollateralScreen', module)
+  .addDecorator((getStory) => <View style={styles.decorator}>{getStory()}</View>)
+  .add('with collateral', () => (
+    <SelectedWalletProvider wallet={mocks.wallet}>
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.stories.tsx:75]: wrap SendProvider

File: apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+          [wallet.primaryTokenInfo.id]: collateralConfig.minLovelace,
+        },
+      },
+    },
+    {
+      onSuccess: (yoroiUnsignedTx) => yoroiUnsignedTxChanged(yoroiUnsignedTx),
+    },
+  )
+
+  const {isLoading: isLoadingCollateral, setCollateralId} = useSetCollateralId(wallet)
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.tsx:60]: Is there any way we can call `setCollateralId` right after we submit the TX? If not, do we have to navigate back to this screen and click "generate collateral" again?

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.tsx:60]: @stackchain ⬆️

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.tsx:60]: Should ```setCollateralId(`${signedTx.signedTx.id}:0`);``` in `onSuccess` in `ConfirmTxScreen` work? I don't quite like the index is hardcoded

Comment (stackchain) [reply apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.tsx:60]: can you read the outputs of the unsigned tx?

Comment (stackchain) [reply apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.tsx:60]: let's create task for these 3 remaining items

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Settings/ManageCollateral/ManageCollateralScreen.tsx:60]: Hello @stackchain 3 remaining items, you mean the following?
1. setCollateralId after tx is completed
2. check copies with product
3. check suggested layout with product

**Pull request merged**